packages/mui-base/src/useSlider/useSlider.types.ts

```diff
@@ -62,3 +63,47 @@ export type UseSliderHiddenInputProps<TOther = {}> = Omit<
   keyof UseSliderHiddenInputOwnProps
 > &
   UseSliderHiddenInputOwnProps;
+
+export type Axis = 'horizontal' | 'vertical' | 'horizontal-reverse';
+
+export interface AxisProps<T extends Axis> {
+  offset: (
+    percent: number,
+  ) => T extends 'horizontal'
+    ? { left: string }
+    : T extends 'vertical'
+    ? { bottom: string }
+    : T extends 'horizontal-reverse'
+    ? { right: string }
+    : never;
+  leap: (
+    percent: number,
+  ) => T extends 'horizontal' | 'horizontal-reverse'
+    ? { width: string }
+    : T extends 'vertical'
+    ? { height: string }
+    : never;
+}
+
+export interface UseSliderReturnValue {
+  active: number;
```

Comment:
The point of the PR is not only to type the return value of the hook, but also to add JSDoc descriptions for each of the props. I had mentioned the same in #36407. Can you add the JSDoc descriptions? You can refer other hooks in MUI Base and add them.

Reply:
Added @ZeeshanTamboli